sdk/eventhub/azure-eventhubs/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -0,0 +1,314 @@
+                    log.warning(
+                        "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                        " has an error during running close(). The exception is %r.",
+                        owner_id, eventhub_name, partition_id, consumer_group_name, err
+                    )
+
+        try:
+            if self._partition_initialize_handler:
+                try:
+                    self._callback_queue.put((self._partition_initialize_handler, partition_context), block=True)
+                except Exception as err:  # pylint:disable=broad-except
+                    log.warning(
+                        "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                        " has an error during running initialize(). The exception is %r.",
+                        owner_id, eventhub_name, partition_id, consumer_group_name, err
+                    )
+            while self._running and not self._threads_stop_flags[partition_id]:
+                try:
+                    events = partition_consumer.receive()
+                    if events:
+                        self._last_enqueued_event_properties[partition_id] = \
+                            partition_consumer.last_enqueued_event_properties
+                        with self._context(events):
+                            self._callback_queue.put((self._event_handler, partition_context, events), block=True)
+                except Exception as error:  # pylint:disable=broad-except
+                    process_error(error)
+                    break
+                    # Go to finally to stop this partition processor.
+                    # Later an EventProcessor(this one or another one) will pick up this partition again.
+        finally:
+            if self._running and self._threads_stop_flags[partition_id]:
+                # Event processor is running but the partition consumer has been stopped.
+                close(CloseReason.OWNERSHIP_LOST)
+            else:
+                close(CloseReason.SHUTDOWN)
+
+            partition_consumer.close()
+
+    def _start(self):
+        """Start the EventProcessor.
+
+        The EventProcessor will try to claim and balance partition ownership with other `EventProcessor`
+        and start receiving EventData from EventHub and processing events.
+
+        :return: None
+
+        """
+        log.info("EventProcessor %r is being started", self._id)
+        ownership_manager = OwnershipManager(self._eventhub_client, self._consumer_group_name, self._id,
+                                             self._partition_manager, self._ownership_timeout, self._partition_id)
+        if not self._running:
+            self._running = True
+            while self._running:
+                try:
+                    checkpoints = ownership_manager.get_checkpoints() if self._partition_manager else None
```

Comment:
I think ownership_manager.get_checkpoints already returns an empty dict if there is no partition manager, so this check might be unnecessary?